### PR TITLE
Fixnamed options pattern

### PR DESCRIPTION
**Pull request by mmatera**
Improves the support for handling options

### CODE REVIEW

File: test/test_evaluation.py [hunk 1]
```diff
@@ -89,10 +89,28 @@ def test_optionvalues():
     assert(result == expected)
 
     result = session.evaluate("OptionValue[F, {a,b, l}]")
-    expected = session.evaluate('{89, 37, OptionValue[l]}')
+    expected = session.evaluate('{89, 37, l}')
     msg = "OptionValue::optnf: Option name l not found."
     assert result == expected, msg
 
+    session.evaluate('Options[f5]:={"a"->12}')
```
Comment (rocky) [inline test/test_evaluation.py:96]: It's a very minor thing, but the reducing (or DRYing) the repeated sequence of 
```
session.evaluate(...
result = ..
expected = ..
assert(result == expected)
```

is exactly what the loop in the other the other methods in this test are doing. 

I'd like to pull this code down though in a couple of days, and try it to understand it in more detail and try it out.

Comment (mmatera) [reply test/test_evaluation.py:96]: Sure, I can contiene working on other stuff in the meantime

**Comment by mmatera** [comment]
@rocky: I included some of your hits to DRY the tests, and also added some other fixes (in particular, to handle attributes in anonymous functions).

**Review by rocky — APPROVED** [review]
LGTM

**Pull request merged**